koku/masu/database/sql/reporting_ocpazure_ocp_infrastructure_back_populate.sql

```diff
@@ -103,3 +103,106 @@ INSERT INTO {{schema | sqlsafe}}.reporting_ocpusagelineitem_daily_summary (
         ocp_azure.pod_labels,
         rp.provider_id
 ;
+
+-- The Unattributed storage cost can not be associated with
+-- a cluster so we insert it separately.
+INSERT INTO {{schema | sqlsafe}}.reporting_ocpusagelineitem_daily_summary (
+    uuid,
+    report_period_id,
+    usage_start,
+    usage_end,
+    cluster_id,
+    cluster_alias,
+    namespace,
+    data_source,
+    node,
+    persistentvolumeclaim,
+    persistentvolume,
+    storageclass,
+    resource_id,
+    pod_labels,
+    volume_labels,
+    source_uuid,
+    infrastructure_raw_cost,
+    infrastructure_project_raw_cost,
+    infrastructure_usage_cost,
+    supplementary_usage_cost,
+    pod_usage_cpu_core_hours,
+    pod_request_cpu_core_hours,
+    pod_limit_cpu_core_hours,
+    pod_usage_memory_gigabyte_hours,
+    pod_request_memory_gigabyte_hours,
+    pod_limit_memory_gigabyte_hours,
+    node_capacity_cpu_cores,
+    node_capacity_cpu_core_hours,
+    node_capacity_memory_gigabytes,
+    node_capacity_memory_gigabyte_hours,
+    cluster_capacity_cpu_core_hours,
+    cluster_capacity_memory_gigabyte_hours,
+    persistentvolumeclaim_capacity_gigabyte,
+    persistentvolumeclaim_capacity_gigabyte_months,
+    volume_request_storage_gigabyte_months,
+    persistentvolumeclaim_usage_gigabyte_months,
+    raw_currency,
+    cost_category_id
+)
+SELECT uuid_generate_v4() as uuid,
+    ocp_azure.report_period_id,
+    ocp_azure.usage_start,
+    ocp_azure.usage_start,
+    ocp_azure.cluster_id,
+    ocp_azure.cluster_alias,
+    ocp_azure.namespace,
+    ocp_azure.data_source,
+    ocp_azure.node,
+    ocp_azure.persistentvolumeclaim,
+    max(ocp_azure.persistentvolume),
+    max(ocp_azure.storageclass),
+    ocp_azure.resource_id,
+    CASE WHEN ocp_azure.data_source = 'Pod'
+        THEN ocp_azure.pod_labels
+        ELSE '{}'::jsonb
+    END as pod_labels,
+    CASE WHEN ocp_azure.data_source = 'Storage'
+        THEN ocp_azure.pod_labels
+        ELSE '{}'::jsonb
+    END as volume_labels,
+    NULL as source_uuid,
```

Comment:
This doesn't seem correct. How can there be no source_uuid?

Reply:
So, in the previous back populate query we use the cluster_id to get the the openshift source uuid.
```
    JOIN {{schema | sqlsafe}}.reporting_ocpusagereportperiod AS rp
        ON ocp_azure.cluster_id = rp.cluster_id
            AND DATE_TRUNC('month', ocp_azure.usage_start)::date  = date(rp.report_period_start)
```

However, since the cost are `unattributed` we can not associate these cloud costs to a cluster. So, I don't know which OCP source uuid to use here.

Reply:
I guess if this cost is not associated with a specific cluster, then there is no source-uuid. Right? How do we ensure all `Storage unattributed` are correctly cleaned up? If we delete an Azure source, is this cost removed?

Reply:
I think this is a good use case to double check. I checked that if we resummarize the data it doesn't double, but we should double check the clean up.

Reply:
I'm not fully sure this matches my understanding of what we are trying to identify. Aren't we looking for the `node-attached` storage? Shouldn't we be able to attribute these costs to a particular node but not broken down to the namespace level?

Reply:
Part of the requirements for Unattributed storage was to track unprovisioned storage, and the design doc uses `(i. e. EBS volume exists but no PV exists yet` as an example. I took this to mean the Azure  managed disks that have been created, but have not been attached to a node or utilized by Persistent Volumes (PVs) in the cluster.

Feel free to correct me if you feel I am working off of the wrong definition for unprovisioned storage; However, I feel like this SQL does capture what I think to be unprovisioned node attached storage.

Reply:
*Recap:*
This review comment prompted a lot of great discussions that lead down a path of us updating and refining how we define Unattributed node based storage. These details have been captured and added to the design doc.